sdk/core/Microsoft.Azure.Core.NewtonsoftJson/src/Serialization/NewtonsoftJsonObjectSerializer.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Reflection;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// A <see cref="NewtonsoftJsonObjectSerializer"/> implementation that uses <see cref="JsonSerializer"/> to for serialization/deserialization.
+    /// </summary>
+    public class NewtonsoftJsonObjectSerializer : ObjectSerializer, ISerializedNameProvider
+    {
+        private const int DefaultBufferSize = 1024;
+
+        // Older StreamReader and StreamWriter would otherwise default to this.
+        private static readonly Encoding UTF8NoBOM = new UTF8Encoding(false, true);
```

Comment:
**Question:** Do you expect that the input stream used to read JSON from and deserialize could contain a BOM? If so, the current solution in the implementation of `JsonObjectSerializer.Deserialize` won't work and will throw if the BOM is detected. You have to explicitly check for and skip the BOM, if one exists since those byte are invalid (as far as JSON is concerned).

Reply:
This implementation is specific to JSON.NET, and the `StreamReader` will detect (i.e. supports) a BOM if present.

Reply:
Yes, I understand. I am pointing out that the STJ specific implementation you have won't work when a BOM is present.

I will add a comment to the correct line, which might be more helpful.